etc/kayobe/ansible/ubuntu-upgrade.yml

```diff
@@ -0,0 +1,103 @@
+  tasks:
+    - name: Assert that hosts are running Ubuntu Focal
+      assert:
+        that:
+          - ansible_facts.distribution == 'Ubuntu'
+          - ansible_facts.distribution_major_version == '20'
+          - ansible_facts.distribution_release == 'focal'
+          - os_distribution == 'ubuntu'
+        fail_msg: >-
+          This playbook is only designed for Ubuntu Focal 20.04 hosts. Ensure
+          that you are limiting it to only run on Focal hosts and
+          os_distribution is set to ubuntu.
+
+    - name: Ensure apt packages are up to date
+      apt:
+        update_cache: true
+        upgrade: yes
+      become: true
+
+    # NOTE: We cannot use apt_repository here because definitions must exist within the standard repos.list
+    - name: Ensure Jammy repo definitions exist in sources.list
+      blockinfile:
+        path: /etc/apt/sources.list
+        block: |
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-updates main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-backports main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_security_url }} jammy-security main restricted universe multiverse
+      become: true
+
+    - name: Ensure do-release-upgrade is installed
+      package:
+        name: ubuntu-release-upgrader-core
+        state: latest
+      become: true
+
+    - name: Do release upgrade
+      command: do-release-upgrade -f DistUpgradeViewNonInteractive
+      become: true
+
+    - name: Ensure Jammy repo definitions do not exist in sources.list
+      blockinfile:
+        path: /etc/apt/sources.list
+        state: absent
+        block: |
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-updates main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-backports main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_security_url }} jammy-security main restricted universe multiverse
+      become: true
+
+    - name: Ensure old venvs do not exist
+      file:
+        path: "/opt/kayobe/venvs/{{ item }}"
+        state: absent
+      loop:
+        - kayobe
+        - kolla-ansible
+      become: true
+
+    - name: Update Python and current user facts before re-creating Kayobe venv
+      ansible.builtin.setup:
+        gather_subset:
+          - '!all'
+          - python
+          - python_version
+          - user_uid
+          - user_gid
+
+- name: Run kayobe-target-venv playbook to ensure kayobe venv exists on remote host
+  import_playbook: "{{ lookup('ansible.builtin.env', 'VIRTUAL_ENV') }}/share/kayobe/ansible/kayobe-target-venv.yml"
+
+- name: Run network playbook to ensure networking config is not reliant on ifupdown
```

Comment:
Ubuntu should never use ifupdown. I wonder why this is necessary.

Reply:
Hosts lost gateway definition after rebooting it to complete the upgrade. So, running the network playbook was necessary to keep the ssh connection with it. I think the name should be updated as it's not ifupdown was issue only when using ubuntu upstream as a source of do-release-upgrade.

Reply:
I've updated the name of the play